cmd/ipfs/main.go

```diff
@@ -416,17 +416,13 @@ func commandShouldRunOnDaemon(details cmdDetails, req cmds.Request, root *cmds.C
 		return nil, err
 	}
 
-	if client != nil { // daemon is running
+	if client != nil { // api file exists
 		if details.cannotRunOnDaemon {
-			e := "cannot use API with this command."
-
 			// check if daemon locked. legacy error text, for now.
-			daemonLocked, _ := fsrepo.LockedByOtherProcess(req.InvocContext().ConfigRoot)
-			if daemonLocked {
-				e = "ipfs daemon is running. please stop it to run this command"
+			if daemonLocked, _ := fsrepo.LockedByOtherProcess(req.InvocContext().ConfigRoot); daemonLocked {
+				return nil, cmds.ClientError("ipfs daemon is running. please stop it to run this command")
 			}
-
-			return nil, cmds.ClientError(e)
+			return nil, nil
```

Comment:
- api file exists does not necessarily imply daemon exists, e.g. the case when the daemon gets killed.
- because https://github.com/ipfs/go-ipfs/blob/83246d3e2d00e7a30ea7d606df9830198f864c4a/cmd/ipfs/main.go#L324 only checks for nil.